testplan/web_ui/testing/src/Common/Home.jsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+
+const NoReport = () => (
+  <h1 style={{ color: "red" }}>NO TEST REPORT TO RENDER</h1>
+);
+
+/**
+ * When NODE_ENV === 'development' we can choose an actual report to render
+ * by setting the environment variable REACT_APP_REPORT_GUID_OVERRIDE
+ * to an actual report GUID.
+ *
+ * Note that process.env.REACT_APP_* vars are elided during compilation. See
+ * node_modules/react-scripts/config/env.js:73.
+ */
+const DevHome = () => {
+  const REPORT_GUID_OVERRIDE = process.env.REACT_APP_REPORT_GUID_OVERRIDE;
+  if(typeof REPORT_GUID_OVERRIDE !== "undefined") {
+    const dest = `/testplan/${REPORT_GUID_OVERRIDE}`;
+    console.log(`REACT_APP_REPORT_GUID_OVERRIDE="${REPORT_GUID_OVERRIDE}" so redirecting to "${dest}"`);
+    return <Redirect to={dest} />;
+  } else {
+    return (
+      <>
+        <NoReport/>
+        <h2 style={{ color: "green" }}>
+          Set these environment variables before compiling:
+          <ul>
+            <li>REACT_APP_REPORT_GUID_OVERRIDE - GUID of an existing report</li>
+            <li>
+              REACT_APP_COUCHDB_HOST - Full 'scheme://host:port/path' of your couchdb server,
```

Comment:
Bear in mind that the UI is not only used for rendering reports from couchdb - it can also be used to render reports generated locally with the --ui flag and for the interactive mode. So I don't think there should be a requirement to specify a couchdb server, it should be possible to run completely locally.

Reply:
Really I wouldn't mention couchdb at all here really - the UI is designed against an API provided by the web_app.py module, which abstracts away the couchdb API.

Reply:
Ya I'm fine with changing the name. This is mostly meant to ease our debugging, and AFAIK there isn't a decent way to debug users' reports without manually downloading their whole json and kludging it into the app in development mode somehow.

Regarding the /interactive route and running locally, there seems to be very little contact between the BatchReport UI and the InteractiveReport UI, aside from sharing some UI components. As they exist right now they can have independent logic behing-the-scenes (though I'm pretty sure this is/was on your TODO, to deduplicate the code between the two reports).

And, while I'm thinking of it, so maybe some future person browsing this PR can remind me, we ought to publish our CouchDB JSON schema so new users can get up and running quicker. We have a log of logic that seems quite specific to how we use it at MS so we should think about either simplifying our schema so it can be inferred by new users or just publishing what we use (or - better - do both).

Reply:
Cool I think changing the name and comment not to mention couchdb would be great.

The batch and interactive UIs share the central assertion pane and most of the toolbar code too. The navigation bar code is different, to accommodate the extra interactive buttons, and the top level report components are significantly different to handle polling for updates from the interactive API instead of downloading the entire batch report as one big JSON. You're right that some of the navigation code could be refactored to avoid duplication in future.

I don't think we currently have a formal schema for the report JSON?

Reply:
No I don't think we do. Appears we've just added incrementally to get what we have currently.

I'd be interested in feedback from any non-MS user who has actually successfully setup a database backend of any sort with Testplan. Our logic in many places is very tailored to the JSON shape we use at present, so anyone looking to setup a database backend would need to just guess from the code what JSON shape the app is expecting. Or monkeypatch their app to use a different JSON shape. Not new-user-friendly ATM.

Reply:
The JSON structure is defined by our testplan code when serializing the report tree, it does not depend on the exact database used to store the data.